### ISSUE TITLE
symbol lookup error: /usr/bin/io.elementary.files: undefined symbol: marlin_sidebar_plugin_item_get_action_group_namespace

### ISSUE THREAD

**Issue by finetuned89** [elementary/files#1085]
After updating to the new 4.2 version, it will no longer start. Starting it from Terminal immediately returns the following error:

```
symbol lookup error: /usr/bin/io.elementary.files: undefined symbol: marlin_sidebar_plugin_item_get_action_group_namespace
```

Compiling from source and installing yields the same result. Same with removing all traces of the app and then reinstalling it from AppCenter.

What can I do to restore the Files app to a working state?

**Comment by tintou** [comment]
I'm guessing that you either have an old plugin leftover (either 3rd party or from a manual installation).
You might want to:
`sudo rm -R /usr/lib/x86_64-linux-gnu/io.elementary.files/plugins/`
then
`sudo apt install pantheon-files --reinstall`

**Comment by finetuned89** [comment]
[Thanks for your suggestion. Unfortunately, the problem remains the same. I tried deleting these files:

```
/usr/include/pantheon-files-core/marlin-file-operations.h
/usr/include/pantheon-files-core/marlin-undostack-manager.h
/usr/local/include/pantheon-files-core/marlin-file-changes-queue.h
```

before reinstalling but that also did not help.

I have a separate elementaryOS installation on a different machine that still works. I tried copying files found using `locate elementary.files` and `locate pantheon-files` to my other machine (and can confirm those files are in the right places), but still the same result.

Could it be that I have some sort of library installed that it gets confused by? FWIW, `sudo apt search --installed | grep marlin` comes up empty.

**Comment by jeremypw** [comment]
The only official sidebar plugin is the new cloudproviders one.  The relevant file is `libpantheon-files-cloudproviders.so`. Make sure you have only one copy of this.  Look under /usr/local/ as well just in case.

**Comment by jeremypw** [comment]
It looks like you have the cloudproviders plugin installed but the version of Files you are running is older than the one where this plugin was incorporated - so it does not recognise the symbols it uses.

**Comment by jeremypw** [comment]
Cloudprovider support was incorporated in version 4.2.0

**Comment by finetuned89** [comment]
Hi jeremypw. I was having this problem with the latest version of Files from AppCenter and from the master branch.

However, your suggestions helped me fix it. I removed `libcloudproviders0` and `libcloudproviders-dev` and all traces of the `libcloudproviders.so.*` modules, as well as all traces of `pantheon-files` and `io.elementary.files`. Then rebooted and reinstalled Files from the AppCenter and it worked again.

(I tried installing files between all these removing steps as well to see if it was solved, but in the end this was the solution).

Thanks.
